packages/solid/src/reactive/signal.ts

```diff
@@ -56,25 +53,21 @@ interface Owner {
   componentName?: string;
 }
 
-interface Computation<T> extends Owner {
-  fn: (v?: T) => T;
+export interface Computation<T extends U, U = T> extends Owner {
+  fn: EffectFunction<T, U>;
   state: number;
   tState?: number;
-  sources: Signal<T>[] | null;
+  sources: SignalState<T>[] | null;
   sourceSlots: number[] | null;
-  value?: T;
+  value?: U;
```

Comment:
The reason for `U` is that value is going to be passed to the function's input parameter, which the return type `T` is assignable to, and the function may transform it into a `T` for the next iteration.

Reply:
In my previous comment this is true about value passed into the effect functions. What about `Computation`s? If those hold only the returned values, then I think what you said makes sense.

Reply:
Just took a look, and it has to be `U` because the initial value is passed to that location, then after that it becomes `T` on second iteration.

Reply:
Oh, oops, I misread what you said. You meant just to remove `?`.

Reply:
I tried removing `?`, and that introduced 5 new errors.

Reply:
My bad, I should've phrased that better. I think it makes sense to make it not optional; if it's a type only returned by `createComputation` and `createComputation` always creates it with some value, then it should always exist, even if undefined. 

There are probably other instances like this where optional properties could be typed ever so slightly better, but they're usually not paid attention to as it's not much of an issue, and in this case especially so since it's on a type meant to be used internally.